tiatoolbox/models/engine/io_config.py

```diff
@@ -0,0 +1,390 @@
+            )
+        else:
+            self.highest_input_resolution = max(
+                self.input_resolutions, key=lambda x: x["resolution"]
+            )
+
+        self._validate()
+
+    def _validate(self):
+        """Validate the data format."""
+        resolutions = self.input_resolutions + self.output_resolutions
+        units = [v["units"] for v in resolutions]
+        units = np.unique(units)
+
+        if len(units) != 1:
+            raise ValueError(
+                f"Multiple resolution units found: `{units}`. "
+                f"Mixing resolution units is not allowed."
+            )
+
+        if units[0] not in [
+            "power",
+            "baseline",
+            "mpp",
+        ]:
+            raise ValueError(f"Invalid resolution units `{units[0]}`.")
+
+    @staticmethod
+    def scale_to_highest(resolutions: List[dict], units: Units):
+        """Get the scaling factor from input resolutions.
+
+        This will convert resolutions to a scaling factor with respect to
+        the highest resolution found in the input resolutions list.
+
+        Args:
+            resolutions (list):
+                A list of resolutions where one is defined as
+                `{'resolution': value, 'unit': value}`
+            units (Units):
+                Resolution units.
+
+        Returns:
+            :class:`numpy.ndarray`:
+                A 1D array of scaling factors having the same length as
+                `resolutions`
+
+        """
+        old_val = [v["resolution"] for v in resolutions]
+        if units not in ["baseline", "mpp", "power"]:
+            raise ValueError(
+                f"Unknown units `{units}`. "
+                "Units should be one of 'baseline', 'mpp' or 'power'."
+            )
+        if units == "baseline":
+            return old_val
+        if units == "mpp":
+            return np.min(old_val) / np.array(old_val)
+        return np.array(old_val) / np.max(old_val)
+
+    def to_baseline(self):
+        """Returns a new config object converted to baseline form.
+
+        This will return a new :class:`ModelIOConfigABC` where
+        resolutions have been converted to baseline format with the
+        highest possible resolution found in both input and output as
+        reference.
+
+        """
+        resolutions = self.input_resolutions + self.output_resolutions
+        save_resolution = getattr(self, "save_resolution", None)
+        if save_resolution is not None:
+            resolutions.append(save_resolution)
+
+        scale_factors = self.scale_to_highest(resolutions, self.resolution_unit)
+        num_input_resolutions = len(self.input_resolutions)
+
+        end_idx = num_input_resolutions
+        input_resolutions = [
+            {"units": "baseline", "resolution": v} for v in scale_factors[:end_idx]
+        ]
+
+        return replace(self, input_resolutions=input_resolutions, output_resolutions=[])
```

Comment:
output_resolutions=[{'units': "baseline", 'resolution': 1.0}]

Reply:
Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/618/commits/a412dd9d7c72adb5bf755815c6b15c1423ecaa24

Reply:
OK, now I remember that this one is a bit weird because the engine design doesn't allow mixed resolution. If we set output_resolution to baseline and input_resolution is in mpp then it won't validate. I think we can try to fix this when we have the new design. For now, I will leave this as empty.

Reply:
https://github.com/TissueImageAnalytics/tiatoolbox/pull/618/commits/766ee7de377fd8e93f4a0bf2767920677adf047d